tools/acorn-optimizer.mjs

```diff
@@ -2107,6 +2107,7 @@ if (!noPrint) {
     keep_quoted_props: closureFriendly, // for closure
     wrap_func_args: false, // don't add extra braces
     comments: true, // for closure as well
+    shorthand: true, // Use object literal shorthand notation
```

Comment:
Won't this wreak havoc with Closure Compiler, like happened with that Web Audio bug?

Reply:
Thats a good question.  I don't think it will do this if string keys are used.  I'll add a test for that.

Reply:
It looks like it does create new shorthand objects, even when the keys are strings.

But even without this change it looks like acorn is removing the string quotes from the keys, which I think would also break closure wouldn't it?   Maybe we just don't run any acorn passes before we run closure?

Reply:
I confirmed that running the acorn_optmizer will remove the quotes from string key today.

I also confirmed the if the quotes a removed from string keys then closure can/will minify them, which will break, for example, pthreads which wants to post message an object with specific string keys.

This means that if we were running closure after acorn today things would already be broken.  Therefore I think this change does not make things worse WRT closure.

Reply:
Oh I see now, we have `keep_quoted_props: closureFriendly, // for closure` in acorn-optimizer.mjs which causes those quotes not to be removed the keys.   Phew, that was confusing.   Mystery solved.   This change does effect the postMessaged objects with quoted keys.

Reply:
> This change does effect the postMessaged objects with quoted keys.

What kind of construct should one use to avoid their postMessage()s from breaking under Closure like this?

Reply:
IIUC the way to stop closure from minifying object field names is to use quotes.    I believe that was true before, and is still true after this change.   i.e. this change doesn't change make things worse for closure users.

Reply:
Yes, you are correct, it will opportunistically apply the shorthand, but we explicit set `keep_quoted_props` when we know the output has to be closure friendly, so it should not effect those quoted properties.